OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearSolverHomotopy.c

```diff
@@ -1571,6 +1572,16 @@ static int newtonAlgorithm(DATA_HOMOTOPY* solverData, double* x)
     {
       solverData->info = 1;
 
+      /* reject new x if old x is as good, for stability (see issue #6419) */
+      if (lastWasGood)
+      {
+        debugString(LOG_NLS_V, "Note: newton solver rejected last x because previous was as good");
+      }
+      else
+      {
+        vecCopy(solverData->n, solverData->x1, x);
```

Comment:
So now `solverData->x1` will only be copied into `x` if  a solution was found instead of updating `x` on every iteration?
I think `x` is used in some way for the damping strategy, so this change disables it?

Reply:
Never mind.... I now saw lines 1646 to 1648.